networkx/algorithms/operators/all.py

```diff
@@ -177,30 +177,24 @@ def compose_all(graphs):
     If a graph attribute is present in multiple graphs, then the value
     from the last graph in the list with that attribute is used.
     """
-    graphs = list(graphs)
+    gen = (graph for graph in graphs)
 
-    if not graphs:
+    try:
+        R = next(gen).copy(as_view=False)
```

Comment:
I had a look at `.__class__()` (before) versus `.copy()` (after), and came to the conclusion that this change is safe, because the copy [source code](https://github.com/networkx/networkx/blob/networkx-2.8.6/networkx/classes/graph.py#L1537) also calls `__class__()` without args (and so fresh attribute dicts get created).

Reply:
cc @dschult (via blame) please correct me if I'm wrong!

Reply:
reverted to `__class__()` in 9e8da58